### ISSUE TITLE
metal3-io/baremetal-operator RAID and Firmware config implementation

### ISSUE THREAD

**Issue by demonCoder95** [airshipit/airshipctl#499]
**Problem description (if applicable)**

**This issue is only to keep track of the upstream metal3-io/baremetal-operator work.**

It is in progress and to be later consumed by airship as part of the RAID/Firmware configuration efforts that have already been added to airshipctl (support for rendering these parameters). Documentation on that is available [here](https://docs.airshipit.org/airshipctl/hardware_profile.html)

The spec work for RAID and BIOS has been merged.
The RAID specification can be found [here](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/raid-api.md)
The BIOS specification can be found [here](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/bios-config.md)

The development efforts are ongoing to get this functionality merged in the upstream metal3-io/baremetal-operator code. The dependency blocking the merger has been only recently resolved. The PRs created for the implementation will be added to the issue later.

**Proposed change**
Implement the RAID and firmware configuration logic in the baremetal operator for the iDRAC BMC type. The airship docs can then use these extended parameters.

**Potential impacts**
None.

**Comment by jezogwza** [comment]
@demonCoder95  can you explain if this means the ability to define raid and bios at the BMH , now translate to a full ETE experience. Meaning BMO, notifies Ironic, and the raid and bios are appropriately configured as per the YAML definition?

**Comment by demonCoder95** [comment]
Yes, that's exactly what this means. As of today, the RAID functionality has been merged upstream and the RAID config is fully functional. We're now working on quickly getting the BIOS functionality merged.
